File: backend/src/main/java/br/com/transoft/backend/dto/vehicle/presenter/VehiclePresenter.java
package br.com.transoft.backend.dto.vehicle.presenter;

import com.fasterxml.jackson.annotation.JsonProperty;

public record VehiclePresenter(
        @JsonProperty("vehicleId") String vehicleId,
        @JsonProperty("plateNumber") String plateNumber,
        @JsonProperty("capacity") Integer capacity,
        @JsonProperty("isActive") Boolean isActive,
        @JsonProperty("vehicleModel") VehicleModelPresenter vehicleModel)
{}

File: transoft-mobile-app/app/model/VehiclePresenter.ts
export default interface VehiclePresenter {
    vehicleId: string;
    plateNumber: string;
    capacity: number;
    isActive: boolean;
    vehicleModel: VehicleModelPresenter;
}

interface VehicleModelPresenter {
    vehicleModelId: string;
    modelName: string;
    modelYear: number;
    automaker: AutomakerPresenter;
}

interface AutomakerPresenter {
    automakerId: string;
    name: string;
}